task-launcher/src/tasks/shared/helpers/replayAudio.ts

```diff
@@ -2,15 +2,17 @@ import { PageAudioHandler } from "./audioHandler";
 import { PageStateHandler } from "./PageStateHandler";
 
 
-export async function setupReplayAudio(pageStateHandler: PageStateHandler) {
+export async function setupReplayAudio(pageStateHandler: PageStateHandler, noAudio: boolean = false) {
```

Comment:
You have the right approach. I am thinking whether we move this behavior to the `PageStateHandler` itself. Because the `PageStateHandler` is responsible for pagestate and `setupReplayAudio` function just listens to pagestate and decides the audio. Potentially we can do something like this:
In PageStateHandler.tsx:

```
constructor(audioFile: string, playStimulusOnLoad: boolean) {
    this.audioFile = audioFile;
    this.audioUri = mediaAssets.audio[camelize(this.audioFile)] ||
    mediaAssets.audio.nullAudio;
    this.getbuffer();
    this.replayBtn = document.getElementById('replay-btn-revisited') as HTMLButtonElement;
    this.playStimulusOnLoad = playStimulusOnLoad !== undefined ? playStimulusOnLoad : true;
  }
```

Initialize the PageStateHandler with the `noAudio` value like you did for `setupReplayAudio`

and then in this function:
```
if (pageStateHandler.replayBtn) {
    if (pageStateHandler.playStimulusOnLoad){
      ... rest of the code
    }
```

Let me know what you think.

Reply:
@zwatson2001 does Aritra's comment make sense? I'd like this one merged soon -- thanks!